### PR TITLE
Add laravel-s for laravel/lumen

### PR DESCRIPTION
**Pull request by hhxsv5**
Hello guys, [laravel-s](https://github.com/hhxsv5/laravel-s) is an out-of-the-box adapter between swoole and laravel, helps developers to use swoole features in laravel conveniently and improve the performance. Thanks.

### CODE REVIEW

File: php/laravel-laravel-s/composer.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "require": {
+    "hhxsv5/laravel-s": "^3.7",
+    "laravel/framework": "^8.12"
```
Comment (waghanza) [inline php/laravel-laravel-s/composer.json:4]: please use `~`

File: php/laravel-laravel-s/composer.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  },
+  "config": {
+    "optimize-autoloader": true,
```
Comment (waghanza) [inline php/laravel-laravel-s/composer.json:7]: not need (this step is done on _containers_) 
https://github.com/the-benchmarker/web-frameworks/blob/e1350bceff5cc31fad0191ab63c55a670d309903/php/Dockerfile#L30

File: php/laravel-laravel-s/composer.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    "preferred-install": "dist",
```
Comment (waghanza) [inline php/laravel-laravel-s/composer.json:8]: what does it mean ?

File: php/laravel-laravel-s/composer.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    "sort-packages": true
```
Comment (waghanza) [inline php/laravel-laravel-s/composer.json:9]: what does it mean ?

File: php/laravel-laravel-s/config.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+environment:
+  APP_ENV: production
+  APP_DEBUG: false
+  APP_KEY: base64:txfHNf/SOo222Rm8I39Urb9SmvUy+nuAF98t/ukF0lk=
+
+before_command:
+  - mkdir bootstrap/cache
+  - mkdir -p storage/logs
+  - php artisan optimize
```
Comment (waghanza) [inline php/laravel-laravel-s/config.yaml:20]: what does it mean ?

File: php/laravel-laravel-s/config/laravels.php [hunk 6]
```diff
@@ -0,0 +1,322 @@
...
+        'log_file'           => storage_path(sprintf('logs/swoole-%s.log', date('Y-m'))),
+        'buffer_output_size' => 2 * 1024 * 1024,
+        'socket_buffer_size' => 8 * 1024 * 1024,
+        'package_max_length' => 4 * 1024 * 1024,
+        'reload_async'       => true,
+        'max_wait_time'      => 60,
+        'enable_reuse_port'  => true,
+        'enable_coroutine'   => false,
+        'upload_tmp_dir'     => @is_writable('/dev/shm/') ? '/dev/shm' : '/tmp',
+        'http_compression'   => false,
```
Comment (waghanza) [inline php/laravel-laravel-s/config/laravels.php:320]: could you use ? 
```suggestion
           'worker_num'       => swoole_cpu_num() * 2,
            'enable_coroutine' => false,
            'log_file' => '/dev/null',
            'log_level' => SWOOLE_LOG_ERROR,
```

File: php/lumen-laravel-s/composer.json [hunk 7]
```diff
@@ -0,0 +1,17 @@
+{
+  "require": {
+    "php": "^7.3|^8.0",
+    "hhxsv5/laravel-s": "^3.7",
+    "laravel/lumen-framework": "^8.0"
```
Comment (waghanza) [inline php/lumen-laravel-s/composer.json:5]: please use `~`

File: php/lumen-laravel-s/composer.json [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+  },
+  "autoload": {
+    "psr-4": {
+      "App\\": "app/"
+    }
+  },
+  "config": {
+    "preferred-install": "dist",
+    "sort-packages": true,
+    "optimize-autoloader": true
```
Comment (waghanza) [inline php/lumen-laravel-s/composer.json:15]: not needed

File: php/lumen-laravel-s/config/laravels.php [hunk 9]
```diff
@@ -0,0 +1,322 @@
...
+    |
+    | Swoole's original configuration items.
+    |
+    | More settings
+    | Chinese https://wiki.swoole.com/#/server/setting
+    | English https://www.swoole.co.uk/docs/modules/swoole-server/configuration
+    |
+    */
+
+    'swoole' => [
```
Comment (waghanza) [inline php/lumen-laravel-s/config/laravels.php:297]: c

File: php/laravel-laravel-s/config/laravels.php [hunk 10]
```diff
@@ -0,0 +1,322 @@
...
+    ],
```
Comment (waghanza) [inline php/laravel-laravel-s/config/laravels.php]: could you use ?
```suggestion
    'swoole' => [
          'worker_num'       => swoole_cpu_num() * 2,
            'enable_coroutine' => false,
            'log_file' => '/dev/null',
            'log_level' => SWOOLE_LOG_ERROR,
    ],
```
then idea is to unify `swoole` config for all `swoole` based **frameworks**

**Review by waghanza — CHANGES_REQUESTED** [review]

**Review by waghanza — CHANGES_REQUESTED** [review]
design think : folders should be name **laravel-s-laravel** / **laravel-s-lumen**, since the framework test is `laravel-s` (not `laravel`)

File: php/laravel-laravel-s/composer.json (hunk 1 above)
Comment (hhxsv5) [reply php/laravel-laravel-s/composer.json:4]: OK

File: php/laravel-laravel-s/composer.json (hunk 2 above)
Comment (hhxsv5) [reply php/laravel-laravel-s/composer.json:7]: OK

**Comment by hhxsv5** [comment]
Modified, please review.

File: php/laravel-s-laravel/composer.json [hunk 11]
```diff
@@ -0,0 +1,11 @@
+{
+  "require": {
+    "hhxsv5/laravel-s": "~3.7",
+    "laravel/framework": "~8.0"
```
Comment (waghanza) [inline php/laravel-s-laravel/composer.json:4]: sorry I had probably misled you. the idea is here to promote compare `laravel-s`. the the important version is `3.7`, no matter in which version `laravel` is
```suggestion
  "require": {
    "hhxsv5/laravel-s": "~3.7.0",
    "laravel/framework": "*"
```

File: php/laravel-s-lumen/composer.json [hunk 12]
```diff
@@ -0,0 +1,11 @@
+{
+  "require": {
+    "hhxsv5/laravel-s": "~3.7",
+    "laravel/lumen-framework": "~8.0"
+  },
```
Comment (waghanza) [inline php/laravel-s-lumen/composer.json:5]:
```suggestion
{
  "require": {
    "hhxsv5/laravel-s": "~3.7.0",
    "laravel/lumen-framework": "*"
  },
```

**Review by waghanza — CHANGES_REQUESTED** [review]
minor changes, are ready to go 🎉

File: php/laravel-s-laravel/composer.json (hunk 11 above)
Comment (hhxsv5) [reply php/laravel-s-laravel/composer.json]: It's okay, I get u.

File: php/laravel-s-laravel/config.yaml [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+  APP_DEBUG: false
+  APP_KEY: base64:txfHNf/SOo222Rm8I39Urb9SmvUy+nuAF98t/ukF0lk=
+
+before_command:
+  - mkdir bootstrap/cache
+  - mkdir storage
+  - php artisan config:cache
+  - php artisan route:cache
+
+command: php bin/laravels start
```
Comment (waghanza) [inline php/laravel-s-laravel/config.yaml:23]: missing file

**Review by waghanza — CHANGES_REQUESTED** [review]

File: php/laravel-s-laravel/config.yaml (hunk 13 above)
Comment (hhxsv5) [reply php/laravel-s-laravel/config.yaml:23]: oh, bin is ignored 😂

**Review by waghanza — APPROVED** [review]

**Comment by hhxsv5** [comment]
@waghanza Why does laravel-s not appear in the [results of 2021-01-26](https://github.com/the-benchmarker/web-frameworks#results-2021-01-26)?

**Pull request merged**